packages/marko/src/runtime/components/index.js

```diff
@@ -10,11 +10,16 @@ var FLAG_HAS_RENDER_BODY = 2;
 var FLAG_IS_LEGACY = 4;
 var FLAG_OLD_HYDRATE_NO_CREATE = 8;
 
-function safeJSONReplacer(match) {
-  if (match === "</") {
-    return "\\u003C/";
-  } else {
-    return "\\u" + match.charCodeAt(0).toString(16);
+function safeStringJSONReplacer(match) {
+  switch (match) {
+    case "'":
+      return "\\'";
+    case "\\":
+      return "\\\\";
+    case "</script":
+      return "\\u003C/script";
+    default:
+      return "\\u" + match.charCodeAt(0).toString(16);
```

Comment:
Does this output leading zeroes if the `charCode` is less than 4 digits? Or do you not need to bother, since the characters it catches will only end up at those higher codepoints anyway?

Reply:
We only escape two characters both with 4 digits [\u2028|\u2029](https://github.com/marko-js/marko/pull/1551/files#diff-55c5215fd78cb23b9a4cb49e164d0066R4). Having said that I think this would be clearer to just include the full escaped sequence, ie:

```
      case "\u2028":
        return "\\u2028";
      case "\u2029":
        return "\\u2029";
```

Reply:
That makes more sense to me upon reading it. Might even be faster too — possibility of a `Map` instead of a `switch` statement too, I think.

Reply:
I haven't benched `Map` vs `switch`, but I could only assume `switch` is faster? Am I wrong?

Reply:
Hmm maybe I'm wrong: https://jsperf.com/if-switch-lookup-table/113 (tested in Chrome). 

Will need to do a more specific bench than this, but if it is similar perf it would make the code a bit nicer I think.

Reply:
Honestly, at this point, I’ve been burnt too many times before on my guesses for what’s faster inside JavaScript engines.

Reply:
Yeah the micro benchmarks lie quite often. Always have to do a real bench. At least for SSR we can get away with just benching a few node versions.

Reply:
Done!